core/clients/key_flow_token_refresh.go

```diff
@@ -0,0 +1,126 @@
+package clients
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+
+	"github.com/stackitcloud/stackit-sdk-go/core/oapierror"
+)
+
+var (
+	defaultTimeStartBeforeTokenExpiration = 30 * time.Minute
+	defaultTimeBetweenContextCheck        = time.Second
+	defaultTimeBetweenTries               = 5 * time.Minute
+)
+
+// Continuously refreshes the token of a key flow, retrying if the token API returns 5xx errrors. Writes to stderr when it terminates.
+//
+// To terminate this routine, close the context in keyFlow.config.TokenRefreshInBackgroundContext.
+func refreshToken(keyflow *KeyFlow) {
+	refresher := &tokenRefresher{
+		keyFlow:                        keyflow,
+		timeStartBeforeTokenExpiration: defaultTimeStartBeforeTokenExpiration,
+		timeBetweenContextCheck:        defaultTimeBetweenContextCheck,
+		timeBetweenTries:               defaultTimeBetweenTries,
+	}
+	err := refresher.refreshToken()
+	fmt.Fprintf(os.Stderr, "Token refreshing terminated: %v", err)
```

Comment:
our SDK doesn't print anything if Im not mistaken, this would be the first thing
I think it would be better to have this method return error, even if we don't do anything with it for now. Just thinking that maybe in the future there could be some extension where this would go into logs

Reply:
Yes, this would be the first thing to be printed

Making this return an error doesn't help. This is called on a background async routine, that can happen at any time, and the routine just terminates when this happens.

The only options I can think of are to do nothing, print to console or add somewhere in the context. The third is useless I think (no one is going to check it, there's no obvious time to when to check it). And printing is better than having it not do anything, but we can suppress this print